### ISSUE TITLE
Support 8x32 row reductions for Kimi V3 FlashMLA

### ISSUE THREAD

**Issue by brnorris03** [tenstorrent/tt-lang#826]
## Required by

- Kimi V3 repository: `tenstorrent/tt-lang-ops-and-models`
- Kimi V3 source checkpoint: [`0c1163f`](https://github.com/tenstorrent/tt-lang-ops-and-models/commit/0c1163fd31bfb24b44db00bf6a3d6e15712319e1)
- Exact released-K3 FlashMLA requirement and reproducer: [`87fbbc1`](https://github.com/tenstorrent/tt-lang-ops-and-models/commit/87fbbc17361787af816be0582f2940b7fde4e348)
- Workload: one `8x192` BF16 query, eight 64-token K/V workers, `8x32` physical query/score pages, and a three-level online-softmax reduction tree.

## Problem

The generated released-K3 FlashMLA operation lowers the scaled `8x32 @ 32x64` query-key matmul, then target validation rejects the first row-wise `ttl.math.reduce_max`:

```text
'ttl.reduce' op tile shape 8x32 is not supported by this compute primitive;
short-height tiles are supported by elementwise, fill, and matmul compute primitives
```

The same operation also requires row-wise `ttl.math.reduce_sum` on an `8x32` probability block. Padding to 16 rows executes twice the released workload and is not an equivalent implementation.

PR #803 added validated short-height elementwise, fill, exponential, and matmul support. Current target validation still intentionally rejects reductions on `8x32` tiles in `test/ttlang/Conversion/TTLToCompute/subtile_invalid.mlir`.

The existing `test/python/test_flash_chain_8node.py` does not cover this requirement. It uses 32 query rows and full `32x32` score tiles; the released K3 operation has eight query rows and `8x32` score tiles.

## Reproducer

At the required-by repository SHA:

```bash
python -m benchmarks.operations.registry run flash_mla
```

The diagnostic runner preserves the initial generated MLIR and verifies the rejection. Run metadata is recorded in `benchmarks/operations/suites/flash_mla/RESULTS.md`.

## Acceptance criteria

- Lower row-wise `ttl.math.reduce_max` and `ttl.math.reduce_sum` correctly for physical `8x32` tiles on Blackhole.
- Preserve the `8x32` result tile contract required by the following broadcast, exponential, elementwise, reciprocal, and matrix operations.
- Add device pytest coverage before updating MLIR tests. Cover `reduce_max` and `reduce_sum`, BF16 and FP32, and supported DRAM/L1 input configurations with dtype-appropriate tolerances.
- Add positive lowering tests for `8x32` reduction and retain negative coverage for unsupported tile dimensions.
- Compile and validate the exact released-K3 FlashMLA reproducer without padding or tensor copies.
- Validate the generated operation against the PyTorch SDPA oracle before collecting performance measurements.
